independent-projects/arc/runtime/src/main/java/io/quarkus/arc/ClientProxy.java

```diff
@@ -3,12 +3,53 @@
 /**
  * Represents a client proxy - a contextual reference to a bean with a normal scope.
  *
- * @author Martin Kouba
+ * <p>
+ *
+ * Client proxy delegates all method invocations to a target contextual instance. Note that only method invocations are
+ * delegated. So a field of a normal scoped bean should never be read or written in order to avoid working with non-contextual
+ * or stale data.
+ *
+ * <p>
+ *
+ * In general, client proxies allow for:
+ * <ul>
+ * <li>Lazy instantiation - the instance is created once a method is invoked upon the proxy.</li>
+ * <li>Ability to inject a bean with "narrower" scope to a bean with "wider" scope; i.e. you can inject a {@code @RequestScoped}
+ * bean into an {@code @ApplicationScoped} bean.</li>
+ * <li>Circular dependencies in the dependency graph. Having circular dependencies is often an indication that a redesign should
+ * be considered, but sometimes it’s inevitable.</li>
+ * <li>In rare cases it’s practical to destroy the beans manually. A direct injected reference would lead to a stale bean
+ * instance.</li>
+ * </ul>
  */
 public interface ClientProxy {
 
+    /**
+     *
+     * @return the contextual instance
+     */
     Object arc_contextualInstance();
 
+    /**
+     *
+     * @return the bean
+     */
     InjectableBean<?> arc_bean();
 
+    /**
+     * This method should only be used with caution. If you unwrap a client proxy then certain key functionality will not work
```

Comment:
The description should mention that in case of beans without proxy, it just returns back the same instance. I am aware that this is the `@return` part, but I'd prefer it here as well :)